### PR TITLE
[Aptos Framework][Staking] Remove max stake flooring and update rewards distribution to mint to both active/pending_inactive

### PR DESCRIPTION
**Pull request by movekevin**
### Description
1. Removing the max stake cap in voting power and rewards distribution. Rationale: We already prevent people from adding stake if the total exceeds the maximum, rewards alone would not take someone that far past the max stake (let's say 10% per year. So it should not affect the network's security/decentralization if we don't "cap" the voting power and reward distribution by the max stake amount.
2. Mint rewards to both active and pending_inactive. This would allow users to receive their rewards earlier instead of auto-locking it in the active stake portion. This also makes the UX of staking services better as they can allow users to unlock funds and still receive the rewards on that unlocked funds until the end of the lockup cycle. For example, if a user requests to withdraw 50 coins half way through their 1-month lockup cycle, they will still be receiving rewards on those 50 coins until the end of the month. This is fair as their 50 coins are not withdrawable and still part of consensus until the end of the month.

### Test Plan
Unit tests

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/2667)

### CODE REVIEW

File: aptos-move/framework/aptos-framework/sources/stake.move [hunk 1]
```diff
@@ -893,11 +893,6 @@ module aptos_framework::stake {
                 continue
             };
 
-            // Restrict a validator's voting power to the max stake allowed.
-            if (new_validator_info.voting_power > maximum_stake) {
-                new_validator_info.voting_power = maximum_stake;
-            };
-
```
Comment (davidiw) [inline aptos-move/framework/aptos-framework/sources/stake.move:900]: I don't know why, this is pretty trivial logic.

**Review by zekun000 — COMMENTED** [review]
I think keeping the cap ensures the consistency of min/max requirement, otherwise the semantic is confusing, probably should rename to maximum_stake_to_join. anyway I don't have strong opinion

File: aptos-move/framework/aptos-framework/sources/stake.move (hunk 1 above)
Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/stake.move:900]: The complexity is if we need to mint to both active and pending_inactive, the code would become a bit more complex to make sure we don't mint rewards to more than max stake amount allowed.

Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/stake.move:900]: When active + pending_inactive > max_stake, we'd need to:
1. Make sure the total rewards minted <= max rewards for max stake.
2. Divide the rewards minted proportionally between active and pending_inactive.

File: aptos-move/framework/aptos-framework/sources/stake.move [hunk 2]
```diff
@@ -512,20 +512,20 @@ module aptos_framework::stake {
         let amount = coin::value<AptosCoin>(&coins);
         assert!(amount > 0, error::invalid_argument(EINVALID_STAKE_AMOUNT));
 
+        let stake_pool = borrow_global_mut<StakePool>(pool_address);
+        let (_minimum_stake, maximum_stake) = staking_config::get_required_stake(&staking_config::get());
+        let total_stake = coin::value<AptosCoin>(&stake_pool.active) +
+                          coin::value<AptosCoin>(&stake_pool.pending_active);
+        assert!(total_stake <= maximum_stake, error::invalid_argument(ESTAKE_EXCEEDS_MAX));
```
Comment (zekun000) [inline aptos-move/framework/aptos-framework/sources/stake.move:519]: this doesn't look right, this is only the existing stake, not the stake + amount. so people can add arbitrary amount to existing stake pool.

what's the rationale of moving this above? fail fast?

**Review by zekun000 — CHANGES_REQUESTED** [review]

File: aptos-move/framework/aptos-framework/sources/stake.move (hunk 2 above)
Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/stake.move]: Yeah good call. I also noticed a minor issue with this validation - it does not take into account pending_inactive. Now as pending_inactive is only moved to inactive at the end of the lockup period, a validator can exceed the max for the remaining time. I fixed this issue and added a lot of tests to make sure the max stake amount is respected in all cases.

Comment (zekun000) [reply aptos-move/framework/aptos-framework/sources/stake.move]: I also just realize that if the node join the validator set, (in pending_active), and add more stake, the coins will go into active but not reflect in the voting power

Comment (movekevin) [reply aptos-move/framework/aptos-framework/sources/stake.move]: I'm not following? on_new_epoch recalculates voting power for every validator that goes into the active set. The voting power is calculated as active + pending_inactive so this should already be taken care of for everyone, including pending_active validators?

Comment (zekun000) [reply aptos-move/framework/aptos-framework/sources/stake.move]: oh right it's regenerated, I was confused by `append(&mut validator_set.active_validators, &mut validator_set.pending_active);` and thought it uses the validator info generated when join

File: aptos-move/framework/aptos-framework/sources/stake.move [hunk 3]
```diff
@@ -521,10 +521,11 @@ module aptos_framework::stake {
             coin::merge<AptosCoin>(&mut stake_pool.active, coins);
         };
 
-        let (_minimum_stake, maximum_stake) = staking_config::get_required_stake(&staking_config::get());
-        let total_stake = coin::value<AptosCoin>(&stake_pool.active) +
+        let (_, maximum_stake) = staking_config::get_required_stake(&staking_config::get());
+        let next_epoch_stake = coin::value<AptosCoin>(&stake_pool.active) +
```
Comment (zekun000) [inline aptos-move/framework/aptos-framework/sources/stake.move:525]: the name is more confusing than total_stake though, pending_inactive may become inactive if the lockup expires next epoch

**Review by zekun000 — APPROVED** [review]

**Pull request merged**